### PR TITLE
samv7: add support for complementary PWM output

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
SAMv7 PWM driver supports complementary PWM output if both pins (H and L) are defined and configured. This commit adds a configuration option to configure the complementary L pin.

The pin definition has to be provided by board level support.

## Impact

SAMv7 PWM driver only.

## Testing

Tested on same70-xplained evaluation kit and on a custom board.

### CODE REVIEW

**Comment by acassis** [comment]
@xiaoxiang781216 could you please take a look on this issue (arm-04), seems it was introduced by your commit 54e630e14d7

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 could you please take a look on this issue (arm-04), seems it was introduced by your commit [54e630e](https://github.com/apache/nuttx/commit/54e630e14d7e32d6f81ae79d4e5df3d2fa787ef0)

The false error come from the new version of gcc, please look this PR: https://github.com/apache/nuttx/pull/8598

File: arch/arm/src/samv7/sam_pwm.c [hunk 1]
```diff
@@ -114,25 +115,45 @@ static struct sam_pwm_channel_s g_pwm0_channels[] =
 #ifdef CONFIG_SAMV7_PWM0_CH0
   {
     .channel = 0,
-    .pin     = GPIO_PWMC0_H0,
+    .pin_h     = GPIO_PWMC0_H0,
```
Comment (xiaoxiang781216) [inline arch/arm/src/samv7/sam_pwm.c:118]: let's keep = align with line 117

File: arch/arm/src/samv7/sam_pwm.c [hunk 2]
```diff
@@ -114,25 +115,45 @@ static struct sam_pwm_channel_s g_pwm0_channels[] =
...
+#else
+    .pin_l     = 0,
+#endif
   },
 #endif
 #ifdef CONFIG_SAMV7_PWM0_CH1
   {
     .channel = 1,
-    .pin     = GPIO_PWMC0_H1,
+    .pin_h     = GPIO_PWMC0_H1,
```
Comment (xiaoxiang781216) [inline arch/arm/src/samv7/sam_pwm.c:129]: ditto

File: arch/arm/src/samv7/sam_pwm.c [hunk 3]
```diff
@@ -185,7 +226,6 @@ static struct sam_pwm_s g_pwm1 =
 };
 
 #endif
-
```
Comment (xiaoxiang781216) [inline arch/arm/src/samv7/sam_pwm.c:188]: revert the change

**Review by pkarashchenko — COMMENTED** [review]
I'm not sure how this coexist with channel override. Need to check in the manual if complimentary channels can be overridden separately.

**Comment by michallenc** [comment]
@pkarashchenko From my experience with PWM testing the overwrite as configured now does not affect a complementary pin.

**Comment by xiaoxiang781216** [comment]
the ci broken is fixed by the last master, please rebase your change.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@pkarashchenko do you have more comment?

**Comment by pkarashchenko** [comment]
Ok. Let's merge it. I will check the SAMv7 manual when I have time

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
Let's ignore the ci broken which is fixed by https://github.com/apache/nuttx/pull/8611 and merge this simple change.
